dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/field/FieldHandlerStrategyFactory.java

```diff
@@ -0,0 +1,352 @@
+     * The value could be a number (translated to an unix timestamp as Date)
+     * A string (translated to a Date using the dateFormats)
+     * Or an actual date (which is just set as it is)
+     * Otherwise throws an exception ({@link DotContentletStateException}).
+     * @param contentlet
+     * @param field
+     * @param value
+     * @param dateFormats
+     */
+    public void parseDate(final Contentlet contentlet,
+                                 final Field field,
+                                 final Object value,
+                                 final String... dateFormats) {
+        if (value instanceof Number) { // is a timestamp
+            contentlet.setDateProperty(field.variable(),
+                    DateUtil.convertDate(Number.class.cast(value).longValue()));
+        } else if (value instanceof Date) {
+            contentlet.setDateProperty(field.variable(), (Date) value);
+        } else if (value instanceof String) {
+            if (((String) value).trim().length() > 0) {
+                try {
+                    final String trimmedValue = ((String) value).trim();
+                    contentlet.setDateProperty(field.variable(),
+                            DateUtil.convertDate(trimmedValue, dateFormats));
+                } catch (Exception e) {
+
+                    throw new DotContentletStateException(
+                            "Unable to convert string to date " + value + ", field: " +
+                                    field.variable());
+                }
+            } else {
+
+                contentlet.setDateProperty(field.variable(), null);
+            }
+        } else if (field.required() && value == null) {
+
+            throw new DotContentletStateException(
+                    "Date fields must either be of type String or Date, field: " +
+                            field.variable());
+        }
+    }
+
+
+    private void booleanStrategy(final Contentlet contentlet, final Field field, final Object value) throws DotContentletStateException {
+
+        if (value instanceof Boolean) {
+
+            contentlet.setBoolProperty(field.variable(), (Boolean) value);
+        } else if (value instanceof String) {
+            try {
+                final String auxValue = (String) value;
+                final Boolean auxBoolean =
+                        (auxValue.equalsIgnoreCase("1") || auxValue.equalsIgnoreCase("true")
+                                || auxValue.equalsIgnoreCase("t")) ? Boolean.TRUE
+                                : Boolean.FALSE;
+                contentlet.setBoolProperty(field.variable(), auxBoolean);
+            } catch (Exception e) {
+                throw new DotContentletStateException(
+                        "Unable to set string value as a Boolean");
+            }
+        } else {
+            throw new DotContentletStateException(
+                    "Boolean fields must either be of type String or Boolean");
```

Comment:
Let's add field and/or contentlet information in the error message.

Reply:
For security we have to take care in adding use defined values that reflect in the error messages.  Not saying not to do this but we have to take extra precautions with sanitisation and how the error is used.